### PR TITLE
JSPF-105 Rely on slf4j-api for logging

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: resolver/src/main/java/org/apache/james/jspf/executor/FutureSPFResult.java [hunk 1]
```diff
@@ -76,9 +71,7 @@ public void setSPFResult(SPFSession session) {
                     listener.onSPFResult(this);
                 } catch (Throwable e) {
                     // catch exception. See JSPF-95
-                    if (log != null) {
-                        log.warn("An exception was thrown by the listener " + listener, e);
-                    }
+                    LOGGER.warn("An exception was thrown by the listener " + listener, e);
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/executor/FutureSPFResult.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/executor/StagedMultipleSPFExecutor.java [hunk 2]
```diff
@@ -19,26 +19,26 @@
...
-import java.util.Map;
+import org.slf4j.LoggerFactory;
 
 /**
  * Async implementation of SPFExecutor
  *
  */
 public class StagedMultipleSPFExecutor implements SPFExecutor, Runnable {
-
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(StagedMultipleSPFExecutor.class);
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/executor/StagedMultipleSPFExecutor.java]: FQN?

File: resolver/src/main/java/org/apache/james/jspf/impl/DNSServiceXBillImpl.java [hunk 3]
```diff
@@ -146,7 +144,7 @@ public void setRecordLimit(int recordLimit) {
         }
         try {
 
-            log.debug("Start "+recordTypeDescription+"-Record lookup for : " + request.getHostname());
+            LOGGER.debug("Start {} Record lookup for : {}", recordTypeDescription, request.getHostname());
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/impl/DNSServiceXBillImpl.java]: you replaced a `-` by a ` `

File: resolver/src/main/java/org/apache/james/jspf/impl/DNSServiceXBillImpl.java [hunk 4]
```diff
@@ -161,11 +159,11 @@ public void setRecordLimit(int recordLimit) {
             
             List<String> records = convertRecordsToList(rr);
             
-            log.debug("Found " + (rr != null ? rr.length : 0) + " "+recordTypeDescription+"-Records");
+            LOGGER.debug("Found {} {}-Records", (rr != null ? rr.length : 0), recordTypeDescription);
             return records;
         } catch (TextParseException e) {
             // i think this is the best we could do
-            log.debug("No "+recordTypeDescription+" Record found for host: " + request.getHostname());
+            LOGGER.debug("No {}} Record found for host: {}", recordTypeDescription, request.getHostname());
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/impl/DNSServiceXBillImpl.java]: 2 `}`

File: resolver/src/main/java/org/apache/james/jspf/impl/DefaultTermsFactory.java [hunk 5]
```diff
@@ -83,7 +76,7 @@ private void init() {
             classes = mechs.split(",");
             Class<?>[] knownMechanisms = new Class[classes.length];
             for (int i = 0; i < classes.length; i++) {
-                log.debug("Add following class as known mechanismn: "
+                LOGGER.debug("Add following class as known mechanismn: "
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/impl/DefaultTermsFactory.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/impl/DefaultTermsFactory.java [hunk 6]
```diff
@@ -92,7 +85,7 @@ private void init() {
             classes = mods.split(",");
             Class<?>[] knownModifiers = new Class[classes.length];
             for (int i = 0; i < classes.length; i++) {
-                log.debug("Add following class as known modifier: "
+                LOGGER.debug("Add following class as known modifier: "
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/impl/DefaultTermsFactory.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java [hunk 7]
```diff
@@ -201,13 +200,13 @@ private void initializePositions() {
             }
         }
 
-        if (log.isDebugEnabled()) {
-            log.debug("Parsing catch group positions: Modifiers["
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Parsing catch group positions: Modifiers["
                     + TERM_STEP_REGEX_MODIFIER_POS + "] Qualifier["
                     + TERM_STEP_REGEX_QUALIFIER_POS + "] Mechanism["
                     + TERM_STEP_REGEX_MECHANISM_POS + "]");
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java [hunk 8]
```diff
@@ -201,13 +200,13 @@ private void initializePositions() {
...
             for (int k = 0; k < matchResultPositions.size(); k++) {
-                log
+                LOGGER
                         .debug(k
                                 + ") "
                                 + (matchResultPositions.get(k) != null ? ((TermDefinition) matchResultPositions
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java [hunk 9]
```diff
@@ -311,7 +310,7 @@ public SPF1Record parse(String spfRecord) throws PermErrorException,
                             TERM_STEP_REGEX_MECHANISM_POS);
 
                     result.getDirectives().add(
-                            new Directive(qualifier, (Mechanism) mech, log.getChildLogger(qualifier+"directive")));
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java:314]: aren't you losing a logging facility here?

File: resolver/src/main/java/org/apache/james/jspf/terms/Directive.java [hunk 10]
```diff
@@ -50,10 +53,10 @@ public DNSLookupContinuation checkSPF(SPFSession spfData)
                 } else {
                     spfData.setCurrentResult(qualifier);
                 }
-                
-                log.info("Processed directive matched: " + Directive.this + " returned " + spfData.getCurrentResult());
+
+                LOGGER.info("Processed directive matched: " + Directive.this + " returned " + spfData.getCurrentResult());
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/terms/Directive.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/terms/Directive.java [hunk 11]
```diff
@@ -50,10 +53,10 @@ public DNSLookupContinuation checkSPF(SPFSession spfData)
...
             } else {
-                log.debug("Processed directive NOT matched: " + this);
+                LOGGER.debug("Processed directive NOT matched: " + this);
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/terms/Directive.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/terms/GenericModifier.java [hunk 12]
```diff
@@ -21,32 +21,30 @@
...
 import org.apache.james.jspf.core.exceptions.TempErrorException;
+import org.slf4j.LoggerFactory;
 
 /**
  * This abstract class represent a gerneric modifier
  * 
  */
-public abstract class GenericModifier implements Modifier, ConfigurationEnabled, LogEnabled {
+public abstract class GenericModifier implements Modifier, ConfigurationEnabled {
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(GenericModifier.class);
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/terms/GenericModifier.java]: FQN?

File: resolver/src/main/java/org/apache/james/jspf/terms/GenericModifier.java [hunk 13]
```diff
@@ -21,32 +21,30 @@
...
 
-    protected Logger log;
-
     /**
      * @see org.apache.james.jspf.core.SPFChecker#checkSPF(org.apache.james.jspf.core.SPFSession)
      */
     public DNSLookupContinuation checkSPF(SPFSession spfData) throws PermErrorException,
             TempErrorException, NeutralException, NoneException {
-        log.debug("Processing modifier: " + this);
+        LOGGER.debug("Processing modifier: " + this);
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/terms/GenericModifier.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/terms/GenericModifier.java [hunk 14]
```diff
@@ -21,32 +21,30 @@
...
         DNSLookupContinuation res = checkSPFLogged(spfData);
-        log.debug("Processed modifier: " + this + " resulted in "
+        LOGGER.debug("Processed modifier: " + this + " resulted in "
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/terms/GenericModifier.java]: {} ?

File: resolver/src/main/java/org/apache/james/jspf/terms/MXMechanism.java [hunk 15]
```diff
@@ -125,7 +128,7 @@ public DNSLookupContinuation onDNSResponse(DNSResponse response, SPFSession spfS
 
             String mx;
             while (records.size() > 0 && (mx = records.remove(0)) != null && mx.length() > 0) {
-                log.debug("Add MX-Record " + mx + " to list");
+                LOGGER.debug("Add MX-Record " + mx + " to list");
```
Comment (rouazana) [inline resolver/src/main/java/org/apache/james/jspf/terms/MXMechanism.java]: {} ?

File: resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java [hunk 16]
```diff
@@ -51,23 +58,14 @@
...
-import java.util.Iterator;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-
-import junit.framework.AssertionFailedError;
-import junit.framework.TestCase;
-
 public abstract class AbstractYamlTest extends TestCase {
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(AbstractYamlTest.class);
```
Comment (rouazana) [inline resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java]: FQN?

File: resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java [hunk 17]
```diff
@@ -132,25 +129,16 @@ protected AbstractYamlTest(String name) throws IOException {
     }
 
     protected void runTest() throws Throwable {
-
-        if (log == null) {
-                log = new ConsoleLogger(ConsoleLogger.LEVEL_DEBUG, "root");
-        }
         
-        log.info("Running test: "+getName()+" ...");
+        LOGGER.info("Running test: "+getName()+" ...");
```
Comment (rouazana) [inline resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java:134]: {} ?

File: resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java [hunk 18]
```diff
@@ -229,7 +216,7 @@ public void wire(Object component) throws WiringServiceException {
                 try {
                     verifyResult(next, queries.get(next));
                 } catch (AssertionFailedError e) {
-                    log.getChildLogger(next).info("FAILED. "+e.getMessage()+" ("+getName()+")", e.getMessage()==null ? e : null);
+                    LOGGER.info("FAILED. "+e.getMessage()+" ("+getName()+")", e.getMessage()==null ? e : null);
```
Comment (rouazana) [inline resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java]: {} ?

File: resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java [hunk 19]
```diff
@@ -240,8 +227,7 @@ public void wire(Object component) throws WiringServiceException {
 
     private SPFResult runSingleTest(String testName) {
         Map<String, ?> currentTest = data.getTests().get(testName);
-        Logger testLogger = log.getChildLogger(testName);
-        testLogger.info("TESTING "+testName+": "+currentTest.get("description"));
+        LOGGER.info("TESTING "+testName+": "+currentTest.get("description"));
```
Comment (rouazana) [inline resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java]: {} ?

File: resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java [hunk 20]
```diff
@@ -296,7 +281,7 @@ private void verifyResult(String testName, SPFResult res) {
     
         }
     
-        testLogger.info("PASSED. Result="+resultSPF+" Explanation="+res.getExplanation()+" Header="+res.getHeaderText());
+        LOGGER.info("PASSED. Result="+resultSPF+" Explanation="+res.getExplanation()+" Header="+res.getHeaderText());
```
Comment (rouazana) [inline resolver/src/test/java/org/apache/james/jspf/AbstractYamlTest.java]: {} ?

File: resolver/src/test/java/org/apache/james/jspf/LoggingDNSService.java [hunk 21]
```diff
@@ -106,10 +107,10 @@ public void setTimeOut(int timeOut) {
             } else {
                 logBuff.append("getRecords-ret: null");
             }
-            logger.debug(logBuff.toString());
+            LOGGER.debug(logBuff.toString());
             return result;
         } catch (TimeoutException e) {
-            logger.debug("getRecords(" + request.getHostname()
+            LOGGER.debug("getRecords(" + request.getHostname()
```
Comment (rouazana) [inline resolver/src/test/java/org/apache/james/jspf/LoggingDNSService.java]: {} ?

File: resolver/src/test/resources/logback-test.xml [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+<configuration>
+        <appender name="FILE" class="ch.qos.logback.core.FileAppender">
+                <file>test-run.log</file>
+                <append>false</append>
+                <!-- set immediateFlush to false for much higher logging throughput -->
+                <immediateFlush>false</immediateFlush>
+                <!-- encoders are assigned the type
+                     ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+                <encoder>
+                        <pattern>bloop %d{HH:mm:ss.SSS} [%-5level] %logger{35} - %msg%n%rEx</pattern>
```
Comment (rouazana) [inline resolver/src/test/resources/logback-test.xml]: bloop? I'm more plop.

File: resolver/src/test/resources/logback-test.xml [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+                </encoder>
+        </appender>
+
+
+        <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+                <encoder>
+                        <pattern>bloop %d{HH:mm:ss.SSS} [%-5level] %logger{15} - %msg%n%rEx</pattern>
```
Comment (rouazana) [inline resolver/src/test/resources/logback-test.xml]: bloop?

File: resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java (hunk 9 above)
Comment (chibenwa) [reply resolver/src/main/java/org/apache/james/jspf/parser/RFC4408SPF1Parser.java:314]: The "directive" part here will be highlighted in the FQCN, I don't think we loose anything.

Just do it the standard way...

File: resolver/src/test/resources/logback-test.xml (hunk 23 above)
Comment (chibenwa) [reply resolver/src/test/resources/logback-test.xml]: Copy pasted from james-project/testing-base. No idea wha&t it means, I trust @mbaechler .

**Comment by chibenwa** [comment]
I must admin I got lazy on the placeholder thing. Would you really consider it blocking for merging that work?

File: resolver/src/test/resources/logback-test.xml (hunk 22 above)
Comment (chibenwa) [reply resolver/src/test/resources/logback-test.xml]: Copy pasted from james-project/testing-base. No idea wha&t it means, I trust @mbaechler .

Comment (rouazana) [reply resolver/src/test/resources/logback-test.xml]: it's clearly a debug statement

File: resolver/src/test/resources/logback-test.xml (hunk 23 above)
Comment (rouazana) [reply resolver/src/test/resources/logback-test.xml]: it's clearly a debug statement

**Comment by rouazana** [comment]
> I must admin I got lazy on the placeholder thing. Would you really consider it blocking for merging that work?

No, but as you did the main work on this, I thought you'd like to finish it.

**Review by rouazana — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**